src/wasm-type.h

```diff
@@ -217,15 +224,24 @@ class HeapType {
   // Get the index of this non-basic type within its recursion group.
   size_t getRecGroupIndex() const;
 
-  constexpr TypeID getID() const { return id; }
-
   // Get the shared or unshared version of this basic heap type.
   constexpr BasicHeapType getBasic(Shareability share) const {
     assert(isBasic());
     return BasicHeapType(share == Shared ? (id | SharedMask)
                                          : (id & ~SharedMask));
   }
 
+  constexpr HeapType with(Exactness exactness) const {
+    assert((!isBasic() || exactness == Inexact) &&
+           "abstract types cannot be exact");
+    return HeapType(exactness == Exact ? (id | ExactMask) : (id & ~ExactMask));
+  }
+
+  constexpr TypeID getID() const { return id; }
+  constexpr TypeID getRawID() const {
```

Comment:
Please add a comment explaining the difference.

And, in particular - when would `getID()` still be used?

Reply:
For example when hashing a type or in the C API to convert the type to an integer.

Reply:
Added the comment.